### ISSUE TITLE
GetPages with optional page count (skip, limit)

### ISSUE THREAD

**Issue by agravity-philipp** [UglyToad/PdfPig#563]
Hi, I have a feature request:

Would it be possible to include a limit on how many pages I want to extract? For example if I just want the first 50 pages of an document I have to load the whole PDF and than interate over the pages and stop.
This takes on a big PDF forever, because it loads all pages.

**Comment by BobLd** [comment]
@agravity-philipp thanks for the question. Can you be more specific about how you load the pages? As far as I know, all pages are not loaded when opening a pdf document.

**Comment by agravity-philipp** [comment]
Basically I try to read out all the texts and save it to a string (and later to a blob).
```
var builder = new StringBuilder();

using (PdfDocument document = PdfDocument.Open(content))
{
    foreach (var page in document.GetPages())
    {
        builder.AppendJoin(" ", page.Text);
    }
}

return builder.ToString();
```

**Comment by BobLd** [comment]
@agravity-philipp `GetPages()` will not load all the pages, it's returning an `IEnumerable<Page>` and won't do anything unless you actually iterate it.

If you don't want to do anything with the first 10 pages and only want the 20 pages after, you can do as follow:
```sharp
foreach (var page in document.GetPages().Skip(10).Take(20))
```

or you can also use `GetPage(x)` to only open the page number `x`, the other pages will not be loaded

**Comment by agravity-philipp** [comment]
@BobLd thanks for clarification. The PDF has nearly 1k pages so it takes a while to read out the whole text... what there be a better approach to get all text?

**Comment by BobLd** [comment]
If speed is your concern, make sure you use `PdfDocument.Open(byte[])` or `PdfDocument.Open(string)`, and *not* `PdfDocument.Open(stream)`, which is much slower.

Also, you could do your `foreach (var page in document.GetPages())` in parallel I think

**Comment by BobLd** [comment]
You can't have both, you need to Mae a choice here

**Comment by agravity-philipp** [comment]
Thats true. Thanks.
